### PR TITLE
Add support to config secret engine, which nosub-path after config

### PR DESCRIPTION
**Pull request by alantang888**
Like "ad", "alicloud", "azure", "gcp", "gcpkms", "kv" secret engine

### CODE REVIEW

File: pkg/vault/operator_client.go [hunk 1]
```diff
@@ -37,6 +37,9 @@ import (
 // DefaultConfigFile is the name of the default config file
 const DefaultConfigFile = "vault-config.yml"
 
+// Secret engine config no need name
+var secretEngineConfigNoNeedName = [...]string{"ad", "alicloud", "azure", "gcp", "gcpkms", "kv"}
```
Comment (bonifaido) [inline pkg/vault/operator_client.go]: Could we use a `map[string]bool` here instead of the slice?

Comment (alantang888) [reply pkg/vault/operator_client.go]: Then use `_, ok := secretEngineConfigNoNeedName[secretEngine]` to check? Value just a dummy value?

Comment (bonifaido) [reply pkg/vault/operator_client.go]: Yes, it is just a dummy value.

Comment (alantang888) [reply pkg/vault/operator_client.go]: Done, just tested with simple config is ok.

File: pkg/vault/operator_client.go [hunk 2]
```diff
@@ -971,3 +979,14 @@ func getMountConfigInput(secretEngine map[string]interface{}) (api.MountConfigIn
 
 	return mountConfigInput, nil
 }
+
+func isConfigNoNeedName(secretEngineType string, configOption string) bool {
+	if configOption == "config" {
+		_, ok := secretEngineConfigNoNeedName[secretEngineType]
+		if ok {
```
Comment (bonifaido) [inline pkg/vault/operator_client.go]: You can simply `return ok` here instead of the `if`.

Comment (alantang888) [reply pkg/vault/operator_client.go]: oh right!! that's redundant...

**Pull request merged**